tsl/src/nodes/vector_agg/functions.c

```diff
@@ -45,29 +45,12 @@ int4_sum_vector(ArrowArray *vector, uint64 *filter, Datum *agg_value, bool *agg_
 	Assert(vector->length <= INT_MAX);
 
 	int64 batch_sum = 0;
-
-	/*
-	 * This loop is not unrolled automatically, so do it manually as usual.
-	 * The value buffer is padded to an even multiple of 64 bytes, i.e. to
-	 * 64 / 4 = 16 elements. The bitmap is an even multiple of 64 elements.
-	 * The number of elements in the inner loop must be less than both these
-	 * values so that we don't go out of bounds. The particular value was
-	 * chosen because it gives some speedup, and the larger values blow up
-	 * the generated code with no performance benefit (checked on clang 16).
-	 */
-#define INNER_LOOP_SIZE 4
-	const int outer_boundary = pad_to_multiple(INNER_LOOP_SIZE, vector->length);
-	for (int outer = 0; outer < outer_boundary; outer += INNER_LOOP_SIZE)
+	for (int row = 0; row < vector->length; row++)
 	{
-		for (int inner = 0; inner < INNER_LOOP_SIZE; inner++)
-		{
-			const int row = outer + inner;
-			const int32 arrow_value = ((int32 *) vector->buffers[1])[row];
-			const bool passes_filter = filter ? arrow_row_is_valid(filter, row) : true;
-			batch_sum += passes_filter * arrow_value * arrow_row_is_valid(vector->buffers[0], row);
-		}
+		const int32 arrow_value = ((int32 *) vector->buffers[1])[row];
+		batch_sum += arrow_value * arrow_row_is_valid(filter, row) *
+					 arrow_row_is_valid(vector->buffers[0], row);
 	}
-#undef INNER_LOOP_SIZE
```

Comment:
The old code noted that the loop was not unrolled. Did that change? If so, since what version of the compilers? You should probably add a comment about that since it would clarify the situation when compiled with an old version of the compiler.

Reply:
I just decided to remove this optimization to simplify things, since it doesn't make a difference for the queries we currently have in the benchmark. I think it makes sense to revisit this when we start optimizing the `sum` function, because there are surprisingly many implementations possible for it, and this unrolled loop is far from the most optimal anyway. At some other DB engine we benchmarked like twenty of them to get close to a competitor's performance.

Reply:
I'm fine with removing this optimization for now. We can start with some more serious tweaking later, but it makes sense to add this information to the commit for posterity.

Reply:
Added a comment.